### PR TITLE
[iOS] Fix: memory leak when Navigation.RemovePage is used

### PR DESCRIPTION
**Pull request by kicsiede**
The leak occurs in a MasterDetailPage.Detail = NavigationPage(ContentPage) scenario if the ContentPage has ToolBarItems. (ToolBarItems create a strong dependency therefore the ViewController of the NavigationPage must be explicitly disposed)

### Description of Change ###

disposing viewcontroller in RemovePage.

### Platforms Affected ### 

- iOS

### PR Checklist ###

- [x] Has automated tests 
- [x] Rebased on top of the target branch at time of PR
- [x] Changes adhere to coding standard

### CODE REVIEW

**Comment by andreinitescu** [comment]
@kicsiede Just a suggestion, adding an unit test for this would be great...

**Comment by rmarinho** [comment]
Hey do you think you can you add a UITest ?! 

thanks

**Comment by kicsiede** [comment]
@rmarinho please check

File: Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue5695.cs [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+#if UITEST
+using NUnit.Framework;
+using Xamarin.UITest;
+using Xamarin.Forms.Core.UITests;
+#endif
+
+namespace Xamarin.Forms.Controls.Issues
+{
+	[Preserve(AllMembers = true)]
+	[Issue(IssueTracker.Github, 5555, "Memory leak when SwitchCell or EntryCell", PlatformAffected.iOS)]
```
Comment (paymicro) [inline Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue5695.cs]: copy/paste detected =)
I'll think about something more universal for any controls when checking for memory leaks.

Comment (rmarinho) [reply Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue5695.cs]: I agree, I was gonna say we had a test for most of the renderers where we pushed a page with a control, and make sure it was collected, I wonder if we can do that in the dynamic gallery test you created Pavel

Comment (kicsiede) [reply Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue5695.cs]: @paymicro @rmarinho fyi here it is not the control causing the leak but the RemovePage method.

**Pull request merged**